### PR TITLE
Fixes #251 Specify DOCPATH and Branch from UI and CI

### LINKED ISSUE #251: Provide option to specify branch and docpath in the webUI
### This is a
- [ ] Bug report
- [x] Feature request

### Parent Issue
#250 

### Current behaviour

Currently if a user wants to use another value for DOCPATH instead of the default `docs/`, they would need to create a `.yaydoc.yml` file in their repo.

### Expected behaviour

Quoting @imujjwal96 from official gitter channel of Yaydoc

> From the perspective of the WebUI, i think we can add two additional (ADVANCE SETTINGS) inputs which are hidden by default. These two can be the branch name and docpath
with the default value as Default branch and docs/ respectively

### PR DESCRIPTION
**Pull request by imujjwal96**
## Related Issue

Fixes #251 
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Sample Deployment
Test this as http://yaydoc6.herokuapp.com/

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@niranjan94  @pri22296 @sch00lb0y instead of registering a branch from the CI, what if at each push to any of the repository's branches, we check if that branch contains a `.yaydoc.yml` file or not. If it does, then we will proceed with the documentation build, otherwise not.

**Comment by sch00lb0y** [comment]
good idea but we can't expect users to have `.yaydoc.yml`. `.yaydoc.yml` is used only if users need some advance configuration. There may be users just want to generate documentation just for readme.md file alone.

**Comment by imujjwal96** [comment]
if a user wants to generate documentation for a README file only, then also he'd have to do the extra work of creating a docs folder as we are assuming the default docpath as `docs/`. If there is no such directory present, then the documentation will not be built.

My point here is if a user has to do some work, why not it'd be the addition of a `.yaydoc.yml` file than creating a directory

**Comment by pri22296** [comment]
Well that's how travis does it, right? I don't see any reason why we also can't go the same route.

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y  Please review.

**Comment by pri22296** [comment]
screenshot 180

I am getting an error with these input. Note that for DOCPATH, I have specified `"."`

**Comment by imujjwal96** [comment]
@pri22296 Was there an error?

**Comment by pri22296** [comment]
@imujjwal96 No. My URL was incorrect that is why deleted message.

**Comment by pri22296** [comment]
If I only want to specify a branch, do I have to input the docpath too?

**Comment by imujjwal96** [comment]
No, those inputs are optional. You can specify any one, all or none

**Comment by pri22296** [comment]
Ok, yeah tested that. But the UI gives no indication about having some defaults. I feel like I am literally setting branch as `""`. Can something be done about that?

**Comment by imujjwal96** [comment]
@pri22296 what do you suggest?

**Comment by pri22296** [comment]
Would it be possible to obtain the default branch using github api and auto-fill the branch input option and just have a default of `docs` in the docpath section?

**Comment by imujjwal96** [comment]
@pri22296 Please check the updated implementation. Hope the UX is improved

**Comment by pri22296** [comment]
@imujjwal96 Please check the form input field for giturl. Input is very laggy there.

**Comment by imujjwal96** [comment]
@pri22296 Can you check for this in yaydoc.herokuapp.com and see if it is reproduced there as well.

**Comment by pri22296** [comment]
No, I cannot reproduce it there. In your deployment input is fine when I am typing the url for the first time. It begins to lag when I edit it or add something later on

**Comment by pri22296** [comment]
Also specifying susi_server's url as GITURL, the cloning fails.

**Comment by imujjwal96** [comment]
@pri22296 I've updated it.

**Comment by pri22296** [comment]
screenshot 182

Though I'm still using the default docpath from the webUI, but the error message does not makes sense here.

**Comment by imujjwal96** [comment]
@pri22296 what do you suggest?

**Comment by pri22296** [comment]
Ok, there seems to be a greater problem here. That error message means that you are passing `""` when the doc path isn't edited. That's seemed like an incorrect behaviour since the ui displays `docs`. But passing docs would mean that user's docpath specified in `.yaydoc.yml` is overriden everytime which is not a good idea.

This would mean that when a user would click on advanced configuration and would see that `docs` is being used as the docpath but if the `.yaydoc.yml` would have a conflicting value, that would be used instead of `docs`.

Can't we have like two checkboxes one for docpath and one for branch and if the checkbox is clicked, you use the ui's version else pass an empty string for `.yaydoc.yml` to handle?

**Comment by imujjwal96** [comment]
@pri22296 made some change in the ui. Please check if it makes sense now

**Comment by pri22296** [comment]
@imujjwal96 Much better. Can the values only be shown when the checkbox is ticked though? and when unchecked, maybe just display a help message or a hint

**Comment by imujjwal96** [comment]
@pri22296 Updated it

File: generate.sh [hunk 1]
```diff
@@ -48,13 +49,17 @@ else
...
-  print_danger "Failed to Clone. Repository does not exist.\n"
+  print_danger "Failed to Clone. Either the repository or the specified branch in that repository does not exist.\n"
   exit 4
 fi
 
 cd ${UNIQUEID}
 print_log "Repository Cloned Successfully!\n"
 
+if [ -z "$DOCPATH" ]; then
+  print_danger "DOCPATH isn't specified. Default value of 'docs' is used"
```
Comment (pri22296) [inline generate.sh]: If a user doesn't uses the advanced configuration checkbox, then It seems this message will be displayed everytime even though if the user has specified a different path in the `.yaydoc.yml` file. In that case this message will be misleading.

**Comment by imujjwal96** [comment]
@pri22296 updated it

**Pull request merged**